### PR TITLE
CS4014

### PR DESCRIPTION
**Pull request by drwill-ms**
<https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014>
Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the await operator to the result of the call.

### CODE REVIEW

File: iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs [hunk 1]
```diff
@@ -68,51 +57,45 @@ public override void SetMonitor(ITransportMonitor usageMeter)
...
             if (Logging.IsEnabled)
-            {
                 Logging.Exit(this, args.Transport, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(WriteAsync)}");
-            }
 
             return true;
         }
 
-        async Task WriteAsyncCore(TransportAsyncCallbackArgs args)
+        private async Task WriteCoreAsync(TransportAsyncCallbackArgs args)
```
Comment (drwill-ms) [inline iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs:82]: rename and explicitly add accessibility modifier

File: iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs [hunk 2]
```diff
@@ -245,50 +220,40 @@ protected override bool CloseInternal()
...
-                }
 
-                var webSocketState = this.webSocket.State;
-                if (webSocketState != IotHubClientWebSocket.WebSocketState.Closed && webSocketState != IotHubClientWebSocket.WebSocketState.Aborted)
+                IotHubClientWebSocket.WebSocketState webSocketState = _webSocket.State;
+                if (webSocketState != IotHubClientWebSocket.WebSocketState.Closed
+                    && webSocketState != IotHubClientWebSocket.WebSocketState.Aborted)
                 {
-                    this.CloseInternalAsync();
+                    _ = CloseInternalAsync();
```
Comment (drwill-ms) [inline iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs:229]: Should this be awaited? For now, I'm explicitly indicating we don't care about waiting for it.

File: iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs [hunk 3]
```diff
@@ -136,26 +119,22 @@ async Task WriteAsyncCore(TransportAsyncCallbackArgs args)
...
                 if (Logging.IsEnabled)
-                {
-                    Logging.Exit(this, args.Transport, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(WriteAsyncCore)}");
-                }
+                    Logging.Exit(this, args.Transport, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(WriteCoreAsync)}");
             }
         }
 
-        async Task<int> ReadAsyncCore()
+        async Task<int> ReadCoreAsync()
```
Comment (barustum) [inline iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs:129]: do we need to add private access modifier?

File: iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs [hunk 4]
```diff
@@ -245,50 +220,40 @@ protected override bool CloseInternal()
...
                 if (Logging.IsEnabled)
-                {
                     Logging.Enter(this, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(CloseInternalAsync)}");
-                }
 
-                await this.webSocket.CloseAsync().ConfigureAwait(false);
+                await _webSocket.CloseAsync().ConfigureAwait(false);
             }
-            catch (Exception e)
+            catch (Exception e) when (!Fx.IsFatal(e))
```
Comment (barustum) [inline iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs:249]: Before, we were rethrowing when it's a fatal exception. After, we don't attempt to rethrow if it's a fatal exception. Is this expected?

Comment (drwill-ms) [reply iothub/device/src/net451/Common/LegacyClientWebSocketTransport.cs:249]: Now we only catch when it isn't fatal. Meaning, if it is fatal, the exception is not caught.

Before, we'd rethrow if it was fatal.

So unless I've made a grievous error, it should be behaving the same.

**Review by barustum — APPROVED** [review]

**Pull request merged**